### ISSUE TITLE
500 Internal Server Error when clicking on "This Week"

### ISSUE THREAD

**Issue by cesalo** [evilhero/mylar#2162]
Browser shows this:

```
500 Internal Server Error
The server encountered an unexpected condition which prevented it from fulfilling the request.
Traceback (most recent call last):
  File "/app/mylar/lib/cherrypy/_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "/app/mylar/lib/cherrypy/lib/encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/app/mylar/lib/cherrypy/_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/app/mylar/mylar/webserve.py", line 1610, in pullist
    repoll = self.manualpull(weeknumber=weekinfo['weeknumber'],year=weekinfo['year'])
  File "/app/mylar/mylar/webserve.py", line 1897, in manualpull
    weeklypull.pullit()
  File "/app/mylar/mylar/weeklypull.py", line 359, in pullit
    comicnm = issname[1]
IndexError: list index out of range
```
Logs shows this, no errors. Debug is enabled:

```
07-Jan-2019 20:45:29 - INFO    :: mylar.pullist.1609 : CP Server Thread-4 : trying to repopulate to week: 01-2019
07-Jan-2019 20:45:29 - INFO    :: mylar.manualpull.1892 : CP Server Thread-4 : ALT_PULL: 0 PULLBYFILE: True week: 01 year: 2019
07-Jan-2019 20:45:29 - INFO    :: mylar.pullit.41 : CP Server Thread-4 : Weekly pull list present - checking if it's up-to-date..
07-Jan-2019 20:45:29 - INFO    :: mylar.pullit.89 : CP Server Thread-4 : [PULL-LIST] Populating & Loading pull-list data from file
07-Jan-2019 20:45:30 - DEBUG   :: mylar.pullit.112 : CP Server Thread-4 : no substitues.csv file located - not performing substitutions on weekly pull list
07-Jan-2019 20:45:30 - DEBUG   :: mylar.pullit.205 : CP Server Thread-4 : shipdate: 20190109
07-Jan-2019 20:45:30 - DEBUG   :: mylar.pullit.206 : CP Server Thread-4 : today: 20181128
07-Jan-2019 20:45:30 - INFO    :: mylar.pullit.212 : CP Server Thread-4 : Preparing to update to the new listing.
```
```
Mylar Version: master
-- git build 34e540c75f06b773e5c8f0c0605345e3911f48c7.
Python Version : 2.7.13

```
Any assistance is welcome thanks.

**Comment by barbequesauce** [comment]
Due to some back-end changes, your alt-pull setting needs to be changed. Shut down Mylar, edit the config.ini, and change

`alt_pull = 0`

to

`alt_pull = 2`

That should do it.

**Comment by cesalo** [comment]
Excellent, that fixed the issue - Thanks a lot. Is there a wiki or post where this parameter is explained?

**Comment by barbequesauce** [comment]
Not yet... maybe evilhero will find some time to explain it :)

**Comment by evilhero** [comment]
```alt_pull=0``` is the old method that was used to populate the weekly pullist. It relied on one site that had alot of improper titles which resulted in alot of manul fixes and just wasn't very pleasant to look at (all capitals, could not go past the current week, etc).

```alt_pull=2``` is the current and most reliable method to populate the weekly pullist (also the default). It uses an intermediary site that clients poll against which allows more granular control over matches, populating, and generally how the pullist appears in the clients. It allows for direct ID matching as well as being able to go ahead several weeks in addition to a host of other things.

Basically if the intermediary site goes offline or has an sql issue, the weekly pull will appear blank to the clients in which mylar then tries to populate it with something (at least temporarily) and falls back to the ```alt_pull=0``` method. If however you do a config save/update while it's in this mode, it will overwrite the current alt_pull method in your config with the fallback option, thereby creating this problem.

**Comment by cesalo** [comment]
Thanks for the info. Appreciate it.
